File: JavaScript/Coin.js
class Coin extends Sprite{

    constructor(x, y, model){
        super(x,y, 32, 32, model);
        this.vertVel = -15.0;
        this.horVel = (Math.random()*20)-10;
        //this.horVel =4;
        this.model = model;
        this.isCoin = true;
        this.lazyLoad();
    }

    isCoin(){
        return true;
    }

    update(){
        this.vertVel += 2.0;
        this.y += this.vertVel;
        this.lastTouchCounter++;

        //horizontal velocity dampening
        if(this.horVel < .25 && this.horVel > -.25)
            this.horVel = 0.0;
        else
            this.horVel/=1.1;
        this.x += this.horVel;

        //coins stay on ground
        if(this.y + this.image.height > 400){
            this.vertVel = 0.0;
            this.y = 400-this.image.height;
            this.lastTouchCounter = 0;
        }

    } //end coin update

    draw(ctx){
        ctx.drawImage(this.image, this.x - this.model.screenPos, this.y);
    }

    lazyLoad(){
        this.image = new Image();
        this.image.src = "images/coin.png";
    }
}

File: Python/Coin.py
import pygame
import random

#from CoinBlock import *
from Sprite import *

class Coin(Sprite):
    def __init__(self, model, x, y, w, h):  
        super(Coin, self).__init__(x,y,w,h)
        self.w = 30
        self.h = 30
        self.vertVel = -30.0
        self.horVel = self.randomHorVel()
        self.coinImage = pygame.image.load("images/coin.png")
        self.model = model
        self.isCoin = True

    def update(self):
        #gravity and horVel
        self.vertVel += 3.5
        self.y += self.vertVel
        self.x += self.horVel

        #horVel converging to 0
        if self.horVel > .25 or self.horVel < .25:
            self.horVel = self.horVel/1.4
        else:    
            self.horVel = 0.0

        #delets coin if off screen
        if self.y > 600:
            self.model.sprites.remove(self)

    def draw(self, screen):
        screen.blit(self.coinImage, (self.x - self.model.screenPos, self.y))

    def randomHorVel(self):
        return random.randint(-100,100)/5.0